doctor/docs/base.py

```diff
@@ -121,10 +121,101 @@ def get_example_lines(headers: Dict[str, str], method: str, url: str,
     return lines
 
 
+def get_object_reference(obj: Object) -> str:
+    """Gets an object reference string from the obj instance.
+
+    This adds the object type to ALL_RESOURCES so that it gets documented and
+    returns a str which contains a sphinx reference to the documented object.
+
+    :param obj: The Object instance.
+    :returns: A sphinx docs reference str.
+    """
+    resource_name = obj.title
+    if resource_name is None:
+        class_name = obj.__name__
+        resource_name = class_name_to_resource_name(class_name)
+    ALL_RESOURCES[resource_name] = obj
+    return ' See :ref:`resource-{}`.'.format(
+        '-'.join(resource_name.split(' ')).lower().strip())
+
+
+def get_array_items_description(item: Array) -> str:
+    """Returns a description for an array's items.
+
+    :param item: The Array type whose items should be documented.
+    :returns: A string documenting what type the array's items should be.
+    """
+    desc = ''
+    if isinstance(item.items, list):
+        # This means the type has a list of types where each position is
+        # mapped to a different type.  Document what each type should be.
+        desc = ''
+        item_pos_template = (
+            ' *Item {pos} must be*: {description}{enum}{ref}')
+        for pos, item in enumerate(item.items):
+            _enum = ''
+            ref = ''
+            if issubclass(item, Enum):
+                _enum = ' Must be one of: `{}`.'.format(item.enum)
+            elif issubclass(item, Object):
+                ref = get_object_reference(item)
+
+            desc += item_pos_template.format(
+                pos=pos, description=item.description, enum=_enum,
+                ref=ref)
+
+    else:
+        # Otherwise just document the type assigned to `items`.
+        desc = item.items.description
+        _enum = ''
+        ref = ''
+        if issubclass(item.items, Enum):
+            _enum = ' Must be one of: `{}`.'.format(
+                item.items.enum)
+        elif issubclass(item.items, Object):
+            ref = get_object_reference(item)
+        desc = '  *Items must be*: {description}{enum}{ref}'.format(
+            description=desc, enum=_enum, ref=ref)
+    return desc
+
+
```

Comment:
Made this it's own function to clean up the code a bit as it was making `get_json_object_lines` a little long when I added the array specific logic.